operator/pkg/secrets/secrets.go

```diff
@@ -73,15 +76,27 @@ func NewCloudExpander(
 }
 
 // NewCloudExpanderFromAPI creates a new CloudExpander
-func NewCloudExpanderFromAPI(api secrets.SecretAPI) *CloudExpander {
-	return &CloudExpander{client: api}
+func NewCloudExpanderFromAPI(
+	api secrets.SecretAPI,
+	ignoreNotFound bool,
+) *CloudExpander {
+	return &CloudExpander{
+		client:         api,
+		logger:         slog.Default(),
+		ignoreNotFound: ignoreNotFound,
+	}
 }
 
 // Expand expands the secret value by retrieving it from the cloud secret store
 func (t *CloudExpander) Expand(ctx context.Context, name string) (string, error) {
 	value, found := t.client.GetSecretValue(ctx, name)
 	if !found {
-		return "", errors.Newf("secret %s not found", name)
+		err := errors.Newf("secret %s not found", name)
+		if t.ignoreNotFound {
+			t.logger.Warn("cannot expand secret %s, secret not found", slog.Any("error", err))
+			return "", nil
```

Comment:
Should fields that fail to expand be elided instead of turned into empty strings? I image there will be instances where setting an empty string would result in an error rather than side stepping the issue all together.